### PR TITLE
Merge stage to main

### PR DESCRIPTION
**Pull request by IlyaVi**

### CODE REVIEW

**Review by axelrod-blox — APPROVED** [review]

File: src/app/common/stores/applications/SsvWeb/SSV.store.ts [hunk 1]
```diff
@@ -106,8 +108,12 @@ class SsvStore extends BaseStore {
       const clusterData = await getClusterData(getClusterHash(cluster.operators, walletStore.accountAddress), this.liquidationCollateralPeriod, this.minimumLiquidationCollateral);
       const ssvAmount = prepareSsvAmountToTransfer(toWei(amount));
       const tx = await contract.deposit(walletStore.accountAddress, operatorsIds, ssvAmount, clusterData);
-      if (tx.hash) {
-        notifyService.hash(tx.hash);
+      if (tx.hash && walletStore.isContractWallet) {
+        store.dispatch(setIsShowTxPendingPopup(true));
```
Comment (alexstolr) [inline src/app/common/stores/applications/SsvWeb/SSV.store.ts:112]: can move this and 115 outside of if else

File: src/app/common/stores/applications/SsvWeb/SSV.store.ts [hunk 2]
```diff
@@ -170,16 +176,24 @@ class SsvStore extends BaseStore {
           const newBalance = fromWei(cluster.balance) - Number(amount);
           if (getClusterRunWay({ ...process.item, balance: toWei(newBalance) }, this.liquidationCollateralPeriod, this.minimumLiquidationCollateral) <= 0) {
             tx = await contract.liquidate(walletStore.accountAddress, operatorsIds, clusterData);
-            if (tx.hash) {
-              notifyService.hash(tx.hash);
+            if (tx.hash && walletStore.isContractWallet) {
+              store.dispatch(setIsShowTxPendingPopup(true));
```
Comment (alexstolr) [inline src/app/common/stores/applications/SsvWeb/SSV.store.ts:180]: dito

File: src/app/common/stores/applications/SsvWeb/SSV.store.ts [hunk 3]
```diff
@@ -170,16 +176,24 @@ class SsvStore extends BaseStore {
...
             }
             const receipt = await tx.wait();
             const result = receipt.blockHash;
             resolve(result);
           } else {
             tx = await contract.withdraw(operatorsIds, prepareSsvAmountToTransfer(toWei(amount)), clusterData);
-            if (tx.hash) {
-              notifyService.hash(tx.hash);
+            if (tx.hash && walletStore.isContractWallet) {
+              store.dispatch(setIsShowTxPendingPopup(true));
```
Comment (alexstolr) [inline src/app/common/stores/applications/SsvWeb/SSV.store.ts:192]: ditto

File: src/app/common/stores/applications/SsvWeb/Validator.store.ts [hunk 4]
```diff
@@ -206,7 +206,7 @@ class ValidatorStore extends BaseStore {
         }
         const receipt = await tx.wait();
         if (receipt.blockHash) {
-          await executeAfterEvent(async () => !!await getEventByTxHash(receipt.transactionHash), async () => this.refreshOperatorsAndClusters(resolve, true), delay);
+          await executeAfterEvent(async () => !await getValidator(validators[0]), async () => this.refreshOperatorsAndClusters(resolve, true), delay);
```
Comment (alexstolr) [inline src/app/common/stores/applications/SsvWeb/Validator.store.ts:209]: Code duplication with line 173

**Review by alexstolr — APPROVED** [review]

**Review by Tomblox1 — APPROVED** [review]

**Pull request merged**